redfish-core/include/utils/hw_isolation.hpp

```diff
@@ -335,5 +337,338 @@ inline void
+                                        *msgPropVal};
+
+                                    // Fill the "msgPropVal" as reason
+                                    std::string message = msgReg->message;
+                                    int i = 0;
+                                    for (const std::string& messageArg :
+                                         messageArgs)
+                                    {
+                                        std::string argIndex =
+                                            "%" + std::to_string(++i);
+                                        size_t argPos = message.find(argIndex);
+                                        if (argPos != std::string::npos)
+                                        {
+                                            message.replace(argPos,
+                                                            argIndex.length(),
+                                                            messageArg);
+                                        }
+                                    }
+                                    // Severity will be added based on the event
+                                    // object property
+                                    condition["Message"] = message;
+                                    condition["MessageArgs"] = messageArgs;
+                                    condition["MessageId"] =
+                                        "OpenBMC.0.2.HardwareIsolationReason";
+                                }
+                                else if (property.first == "Severity")
+                                {
+                                    const std::string* severity =
+                                        std::get_if<std::string>(
+                                            &property.second);
+                                    if (severity == nullptr)
+                                    {
+                                        BMCWEB_LOG_ERROR
+                                            << "Failed to get the Severity "
+                                            << "from object: "
+                                            << hwStatusEventObj;
+                                        messages::internalError(aResp->res);
+                                        break;
+                                    }
+
+                                    // we have only one condition
+                                    nlohmann::json_pointer severityPropPath =
+                                        "/Status/Conditions/0/Severity"_json_pointer;
+                                    if (!setSeverity(aResp, hwStatusEventObj,
+                                                     severityPropPath,
+                                                     *severity))
+                                    {
+                                        // Failed to set the severity
+                                        break;
```

Comment:
I think should be return which is more common in bmcweb, set an internalError and return

Reply:
Done